packages/react-dom/src/__tests__/ReactDOMServerIntegrationHooks-test.internal.js

```diff
@@ -546,10 +548,12 @@ describe('ReactDOMServerHooks', () => {
         });
         return <Text text={'Count: ' + props.count} />;
       }
-      const domNode = await render(<Counter count={0} />);
-      expect(clearYields()).toEqual(['Count: 0']);
-      expect(domNode.tagName).toEqual('SPAN');
-      expect(domNode.textContent).toEqual('Count: 0');
+      await act(async () => {
+        const domNode = await render(<Counter count={0} />);
```

Comment:
one variant of this test fails "renders should ignore effects on the server with client render on top of good server markup"